src/accordion/tp-accordions-accordion-handle.ts

```diff
@@ -0,0 +1,46 @@
+import { TPAccordionsElement } from './tp-accordions';
+
+/**
+ * TPAccordionsAccordionHandleElement.
+ */
+export class TPAccordionsAccordionHandleElement extends HTMLElement {
+	private accordions: TPAccordionsElement | null;
+	constructor() {
+		super();
+
+		this.accordions = this.closest( 'tp-accordions' );
+	}
+
+	connectedCallback() {
+		this.addEventListener( 'click', () => this.toggle() );
+	}
+
+	/**
+	 * Toggle accordion state.
+	 */
+	toggle() {
+		if ( this.accordions ) {
+			this.accordions.setAttribute( 'expand-all', 'no' );
+			this.accordions.setAttribute( 'collapse-all', 'no' );
+		}
+
+		// Toggle accordion item states.
+		this.parentElement?.toggleAttribute( 'open' );
```

Comment:
@sagarjadhav Lets use `this.closest( 'tp-accordion' )`

Reply:
I believe you mean `this.closest( 'tp-accordion-item' );` @junaidbhura .